src/main/java/org/elasticsearch/river/kafka/KafkaWorker.java

```diff
@@ -100,22 +97,10 @@ private void consumeMessagesAndAddToBulkProcessor(final KafkaStream stream) {
                 final MessageAndMetadata messageAndMetadata = consumerIterator.next();
                 logMessage(messageAndMetadata);
 
-                messageSet.add(messageAndMetadata);
```

Comment:
@mariamhakobyan this was a while ago now, but I believe my logic was that elasticsearch's own bulkprocessor does the exact same thing with its "setBulkActions(int bulkActions)". It looked like we were doing the same thing twice, once with your Set<> and once in Bulkprocessor. Is that not correct?